code/datums/components/cleave_attack.dm

```diff
@@ -0,0 +1,78 @@
+/datum/component/cleave_attack
+	dupe_mode = COMPONENT_DUPE_UNIQUE_PASSARGS
+	var/arc_size = 90 // size of the attack arc in degrees
+	var/requires_wielded = FALSE // make this TRUE for two-handed weapons like axes
+	var/swing_speed_mod = 1.5 // how much slower is it to swing
+	var/cleave_effect = /obj/effect/temp_visual/dir_setting/firing_effect/mecha_swipe
```

Comment:
autodoc please
Also since you have it all in the args in ``Initialize``, you don't need to have these declared here and it can just be empty.